utils/_context/core.py

```diff
@@ -89,28 +92,32 @@ def configuration(self):
         return self._get_scenario_property("configuration", {})
 
     @property
-    def vm_os_branch(self):
-        if not hasattr(self.scenario, "virtual_machine"):
-            return None
-
-        virtual_machine = self.scenario.virtual_machine
-        return virtual_machine.os_branch
+    def virtual_machine(self) -> _VirtualMachine:
+        return self._get_scenario_property(
+            "virtual_machine",
+            _VirtualMachine(
+                name="",
+                aws_config=None,
+                vagrant_config=None,
+                krunvm_config=None,
+                os_type=None,
+                os_distro=None,
+                os_branch="",
+                os_cpu="",
+            ),
+        )
 
     @property
-    def vm_os_cpu(self):
-        if not hasattr(self.scenario, "virtual_machine"):
-            return None
-
-        virtual_machine = self.scenario.virtual_machine
-        return virtual_machine.os_cpu
+    def vm_os_branch(self) -> str:
```

Comment:
I will change this in the near future. Instead of using '''@bug(context.vm_os_branch="xyz" ''' we could use '''@bug(context.virtual_machine.os_branch="xyz" ''